scripts/skills/perks/perk_rf_kingfisher.nut

```diff
@@ -35,82 +36,92 @@ this.perk_rf_kingfisher <- ::inherit("scripts/skills/skill", {
 
 	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
 	{
-		if (!_targetEntity.isAlive())
+		if (!_targetEntity.isAlive() || ::Math.rand(1, 100) > this.m.Chance || !this.getContainer().getActor().isPlacedOnMap() || !_targetEntity.isPlacedOnMap())
 			return;
 
-		if (::Math.rand(1, 100) <= this.m.Chance)
+		local throwNetSkill = this.getContainer().getSkillByID("actives.throw_net");
+		if (!throwNetSkill.onVerifyTarget(this.getContainer().getActor().getTile(), _targetEntity.getTile()))
+			return;
+
+		local netItemScript = ::IO.scriptFilenameByHash(this.getContainer().getActor().getOffhandItem().ClassNameHash);
+
+		throwNetSkill.useForFree(_targetEntity.getTile());
+
+		local replacementNet = ::new(netItemScript)
+
+		this.getContainer().getActor().getItems().equip(replacementNet); // the original net is unequipped during use of throw net skill, so we equip a "new" net of the same type
+		foreach (skill in replacementNet.m.SkillPtrs)
+		{
+			// the skill isn't added to the skill_container yet, because we are within the onTargetHit function, so it goes to SkillsToAdd
+			// so we can't get it via container.getSkillByID, instead we find it by iterating over the skills of the item
+			if (skill.getID() == "actives.throw_net")
+			{
+				skill.m.IsHidden = true;
+				break;
+			}
+		}
+
+		local netEffect = _targetEntity.getSkills().getSkillByID("effects.net");
+		netEffect.m.KingfisherPerk <- ::MSU.asWeakTableRef(this);
```

Comment:
This will probably crash the game when you kingfishered an enemy who is immune to being netted.
You just blindly assume that the net effect is present on the enemy